repository/csv_reader.go

```diff
@@ -0,0 +1,77 @@
+package repository
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"log"
+	"os"
+	"strconv"
+
+	"wizeline/model"
+)
+
+func GetUsersFromCSV(fileName string) (string, error){
+	// open file
+	f, err := os.Open(fileName)
+	if err != nil {
+		log.Println(err)
+		return "", errors.New("Error opening file " + fileName)
+	}
+
+	//close the file at the end of the program
+	defer func(f *os.File) {
+		err := f.Close()
+		if err != nil {
+			log.Println(err)
+		}
+	}(f)
+
+	//Read CSV file using csv.Reader
+	csvReader := csv.NewReader(f)
+	data, err := csvReader.ReadAll()
+	if err != nil {
+		log.Println(err)
+		return "", errors.New("Error reading file " + fileName)
+	}
+
+	userData, err := getUsers(data)
+	if err != nil {
+		log.Println(err)
+		return "", errors.New("Error parsing data " + fileName)
+	}
+	jsonData, err := json.MarshalIndent(userData, "", "  ")
+	if err != nil {
+		log.Println(err)
+		return "", errors.New("Error transforming csv to json " + fileName)
+	}
+	return string(jsonData), nil
+}
+
+func getUsers(data [][]string) ([]model.User, error) {
+	// convert csv lines to array of structs
+	var users []model.User
+	for i, line := range data {
+		if i > 0 { // omit header line
```

Comment:
Instead of having all the logic inside of the `if` sentence, why not just ignore the first line in the if:
```go
if i == 0 {
    continue
}
```